hydrosdk/model.py

```diff
@@ -561,91 +564,67 @@ def __init__(self, name, id_, contract, version, metadata):
         self.id_ = id_
 
 
-class BuildStatus(Enum):
-    """
-    Model building statuses
-    """
-    BUILDING = "BUILDING"
-    FINISHED = "FINISHED"
-    FAILED = "FAILED"
-
-
 class UploadResponse:
     """
     Received status from server about uploading
     """
+
     def __init__(self, model, version_id):
         self.cluster = model.cluster
         self.model = model
         self.model_version_id = version_id
         self.cluster = self.model.cluster
-        self._logs_iterator = self.logs()
-        self.last_log = ""
-        self._status = ""
 
     def logs(self):
         """
         Sends request, saves and returns logs iterator
         :return: log iterator
         """
-        logger = logging.getLogger("ModelDeploy")
-        try:
-            url = "/api/v2/model/version/{}/logs".format(self.model_version_id)
-            logs_response = self.model.cluster.request("GET", url, stream=True)
-            self._logs_iterator = sseclient.SSEClient(logs_response).events()
-        except RuntimeError:
-            logger.exception("Unable to get build logs")
-            self._logs_iterator = None
-        return self._logs_iterator
+        url = "/api/v2/model/version/{}/logs".format(self.model_version_id)
+        logs_response = self.model.cluster.request("GET", url, stream=True)
+        return sseclient.SSEClient(logs_response).events()
 
-    def set_status(self) -> None:
+    def poll_model(self) -> None:
```

Comment:
New realization of logging is much more cumbersome, after each status check request is sent, then object is deserialized and only then status is checked. But I guess  it is required by the task

Reply:
But it's the only correct option here. Parsing log strings for a "Successful" substring doesn't mean that model finished building.